ramandecompy/tests/test_spectrafit.py

```diff
@@ -192,80 +199,93 @@ def test_build_custom_model():
+    fit_result, residuals = spectrafit.build_custom_model(X_TEST, Y_TEST,
+                                                          peaks, peaks_add, plot_fits=True)
     assert len(fit_result) == 4, '4th peak was not successfully added to the model'
     assert isinstance(fit_result, list), '`fit_result` is not a list'
     for i, element in enumerate(fit_result):
-        assert isinstance(element, list), 'element with index {} in fit_result is not a list'.format(i)
+        assert isinstance(element, list), """element with index {}
+        in fit_result is not a list""".format(i)
         if i <= 2:
             assert len(element) == 7, 'index {} in fit_result should have length 7'.format(i)
         else:
             assert len(element) == 8, 'index {} in fit_result should have length 8'.format(i)
-            assert element[-1] == 'user_added', 'user_added tag not successfully appended' 
+            assert element[-1] == 'user_added', 'user_added tag not successfully appended'
     assert len(residuals) == len(X_TEST), 'size of `residuals` does not match input data'
-    assert isinstance(residuals[0], np.float64), 'the 1st element in `residuals` is not a np.float64'
+    assert isinstance(residuals[0], np.float64), """
+    the 1st element in `residuals` is not a np.float64"""
     try:
         spectrafit.build_custom_model(4.2, Y_TEST, peaks, peaks_add, plot_fits=True)
     except TypeError:
-            print('A float was passed to the function, and was handled well with a TypeError.')
+        print('A float was passed to the function, and was handled well with a TypeError.')
     try:
         spectrafit.build_custom_model(X_TEST, 'Y_TEST', peaks, peaks_add, plot_fits=True)
     except TypeError:
-            print('A string was passed to the function, and was handled well with a TypeError.')
+        print('A string was passed to the function, and was handled well with a TypeError.')
     try:
         spectrafit.build_custom_model(X_TEST, Y_TEST, (1, 2, 3, 4, 5, 6), peaks_add, plot_fits=True)
     except TypeError:
-            print('A tuple was passed to the function, and was handled well with a TypeError.')
+        print('A tuple was passed to the function, and was handled well with a TypeError.')
     try:
         spectrafit.build_custom_model(X_TEST, Y_TEST, peaks, 4.2, plot_fits=1)
     except TypeError:
-            print('A float was passed to the function, and it was handled well with a TypeError.')
+        print('A float was passed to the function, and it was handled well with a TypeError.')
     try:
         spectrafit.build_custom_model(X_TEST, Y_TEST, peaks, peaks_add, plot_fits=1)
     except TypeError:
-            print('An int was passed to the function, and was handled well with a TypeError.')
+        print('An int was passed to the function, and was handled well with a TypeError.')
 
 
 def test_apply_old_model():
     """
     docstring
```

Comment:
need doc string here

not sure what this function is...

Reply:
This function is used within the `superimpose_next()` function which is explained in the docstrings in spectrafit itself.